spec/src/main/asciidoc/micrometer-backends.adoc

```diff
@@ -0,0 +1,120 @@
+backends, which at the time of writing include:
+
+* AppOptics
+* Azure Monitor
+* Netflix Atlas
+* CloudWatch
+* Datadog
+* Dynatrace
+* Elastic
+* Ganglia
+* Graphite
+* Humio
+* Influx/Telegraf
+* JMX
+* KairosDB
+* New Relic
+* Prometheus
+* SignalFx
+* Google Stackdriver
+* StatsD 
+* Wavefront
+
+The `/metrics` REST endpoint provides Prometheus metrics as a pull-based mechanism. A monitoring agent will
+need to make a request to the endpoint to obtain the metrics data at that point in time. 
+Conversely, the Micrometer backends listed above are typically push-based, so vendor products using the 
+Micrometer backends will be periodically pushing metrics data from the server to the metrics backend.
+
+=== Recommended setup and configuration for alternative Micrometer backends
+
+The following suggestions are OPTIONAL, and provided with a view of attempting to make configuring
+Micrometer-based metrics implementations consistent for consumers.
+
+==== Discoverability
+
+Each Micrometer backend is packaged separately by the Micrometer project in its own .jar file. In order
+to allow an implementation to push to Graphite, for example, the vendor will either need to provide the 
+io.micrometer:micrometer-registry-graphite jar and its runtime dependencies as part of their product, 
+or enable consumers to add it to their classpath. Where consumers are adding libraries to the classpath,
+vendors can check for the presence of the appropriate MeterRegistry class. Taking Graphite as an example,
+the vendor would need to check for the presence of `io.micrometer.graphite.GraphiteMeterRegistry`.
+
+==== Configuration
+
+Each Micrometer backend has its own Config class, which includes a `String get(final String propertyName)` method.
+In order to configure the Micrometer backends in a way that is both consistent across all the Micrometer backends
+and also consistent with MicroProfile itself, it is suggested that the `String get(final String propertyName)` is
+overridden to obtain the relevant config using MicroProfile config. Micrometer property names are already prefixed
+with the name of the relevant backend, so it is suggested that a prefix of `microprofile.metrics.` is added to the
+property when it is obtained from MicroProfile config.
+
+==== Enabling a backend
+
+Micrometer backends have many values in their config set by default. It is therefore recommended that backends
+are not enabled by default, and enabled by setting `microprofile.metrics.<backend name>.enabled` to `true`, for example:
+
+----
+microprofile.metrics.graphite.enabled = true
+----
+
+==== Example backend setup and configuration
+
+If a vendor is implementing MicroProfile Metrics as a CDI extension, the above can be achieved by registering
+a Producer for a backend, if the relevant Micrometer registry class is available on the classpath.
+
+The following is provided as an example of a CDI producer for the Graphite backend.
```

Comment:
Would it make sense (or would it lengthen this discussion too much) to show the subclassing of `GraphiteConfig` as we're describing above?

Reply:
At present I've left the implementation of `GraphiteConfig` inline, but I could pull that out into its own class if we think that makes it easier to read.

Reply:
No, it's fine as-is. I must have missed that impl tucked in there when I read this earlier.